fdbserver/TLogServer.actor.cpp

```diff
@@ -1159,6 +1159,9 @@ void commitMessages( TLogData *self, Reference<LogData> logData, Version version
 Version poppedVersion( Reference<LogData> self, Tag tag) {
 	auto tagData = self->getTagData(tag);
 	if (!tagData) {
+		if (tag == txsTag || tag.locality == tagLocalityTxs) {
+			return 0;
+		}
```

Comment:
If you peek a Log for a txsTag or tagLocalityTxs, it'd end up here, and return recoveredAt, which would result in a spurious popped version update.